### ISSUE TITLE
Multi-label Metrics return tensor of unexpected shape

### ISSUE THREAD

**Issue by mydkzgj** [pytorch/ignite#1233]
## ❓ Questions/Help/Support

Hi, I am working with Multi-label Metrics such as Accuracy, Precision, Recall. I want to reserve those metrics of every class, so I set "average=False, is_multilabel=True". However, the function return a tensor whose shape is batch_size rather than num_categories.
I find the code in [precision.py](https://github.com/pytorch/ignite/blob/master/ignite/metrics/precision.py),  line 152. I don't know why you transpose the y_pred and y. 
```Python
 elif self._type == "multilabel":
            # if y, y_pred shape is (N, C, ...) -> (C, N x ...)
            num_classes = y_pred.size(1)
            y_pred = torch.transpose(y_pred, 1, 0).reshape(num_classes, -1)
            y = torch.transpose(y, 1, 0).reshape(num_classes, -1)
```
I set "average=False, is_multilabel=True" and expect to obtain a tensor whose shape is num_categories. Mabye I have a misunderstanding on “average”？
Thanks.

**Comment by vfdev-5** [comment]
@mydkzgj AFAIK multi-label metrics can not be computed by label as a match is considered over all labels together. However, if you would like to compute label-wise accuracy, precision, recall you can do this directly by specifying prediction and target. Here is an example of how to do this:

```python
evaluator = ...

# Setup multilabel metrics
val_metrics = {
    "Multilabel Accuracy": Accuracy(is_multilabel=True),
    "Multilabel Precision": Precision(is_multilabel=True, average=True),
    "Multilabel Recall": Recall(is_multilabel=True, average=True),       
}

# Setup label-wise metrics
# -> let's assume that evaluator returns output as (y_pred, y)
def get_single_label_output_fn(c):
    def wrapper(output):
        y_pred, y = output
        return y_pred[:, c], y[:, c]
    return wrapper

for i in range(num_labels):
    for name, cls in zip(["Accuracy", "Precision", "Recall"], [Accuracy, Precision, Recall]):
        val_metrics["{}/{}".format(name, i)] = cls(output_transform=get_single_label_output_fn(i))

evaluator.run(val_data)
```

HTH

**Comment by mydkzgj** [comment]
@vfdev-5 Thank you for your help. Nice script, I have tried it in my evaluator and it works.
 And I also tried another way and it seems to work too.  I made this change in my trainer.py. I tranpose the y_pred and y in transform and I only concern about output of every batch because of '"RunningAverage" used in trainer. (Of course, if batch_num is 1 or batch_num changes, it will raise exception.)
```
        metrics_train = {"avg_total_loss": RunningAverage(output_transform=lambda x: x["total_loss"]),
                         "avg_precision": RunningAverage(Precision(average=False, output_transform=lambda x: (x["scores"].sigmoid().round().transpose(1,0), x["labels"].transpose(1,0)), is_multilabel=True)),
                         "avg_accuracy": RunningAverage(Accuracy(output_transform=lambda x: (x["logits"].sigmoid().round(), x["labels"]), is_multilabel=True)),
                         }
```

**Comment by vfdev-5** [comment]
> And I also tried another way and it seems to work too. I made this change in my trainer.py. I tranpose the y_pred and y in transform

Well, I let you make sure that you compute correct values :)

**Comment by mydkzgj** [comment]
@vfdev-5 thanks again！
